### PR TITLE
Change same elderly checks to be case insensitive when comparing names

### PR DESCRIPTION
**Pull request by nicole-luo-exe**
Currently, 
* `addElderly` and `editElderly` allows elderlies Jessica Lee and jessica lee to be in NurseyBook simultaneously.
* `addTask` and `editTask` throw errors when Jessica Lee is present in NurseyBook but the elderly name entered is 'jessica lee'.

Let's introduce case-insensitive name checking.

### CODE REVIEW

File: src/main/java/nurseybook/model/ModelManager.java [hunk 1]
```diff
@@ -307,18 +310,15 @@ public boolean equals(Object obj) {
...
-                    && filteredElderlies.equals(other.filteredElderlies)
-                    && filteredTasks.equals(other.filteredTasks);
-        }
-        return other.elderlyOfInterest != null
+        boolean areElderlyOfInterestsEqual = elderlyOfInterest == null && other.elderlyOfInterest == null
+                ? true
+                : elderlyOfInterest == null || other.elderlyOfInterest == null
+                    ? false
+                    : elderlyOfInterest.equals(other.elderlyOfInterest);
+        return areElderlyOfInterestsEqual
```
Comment (xiongjya) [inline src/main/java/nurseybook/model/ModelManager.java:318]: Would it be better if you simply check that `elderlyOfInterest` is null and add an additional line in the `equals()` method for `Elderly` to check if the `other` object is `null`?

**Review by xiongjya — CHANGES_REQUESTED** [review]
LGTM! After a few minor changes should be alright.

File: src/main/java/nurseybook/model/ModelManager.java (hunk 1 above)
Comment (nicole-luo-exe) [reply src/main/java/nurseybook/model/ModelManager.java:318]: Ah thanks, was trying to find out a better way for this

**Review by xiongjya — APPROVED** [review]
LGTM! 👍

**Pull request merged**